connectivity/netsocket/include/netsocket/nsapi_types.h

```diff
@@ -406,6 +408,23 @@ typedef struct nsapi_stagger_req {
     uint16_t stagger_rand;  /* [OUT] Randomized stagger value in seconds */
 } nsapi_stagger_req_t;
 
+/** nsapi_msghdr
+ */
+typedef struct MBED_ALIGN(double) nsapi_msghdr {
```

Comment:
@kjbracey-arm What is your preference for this alignment ?

Reply:
Actually, C++11 provide a `max_align_t` type (https://en.cppreference.com/w/c/types/max_align_t). Let's use this instead.

Reply:
Done